test/stdlib/CodableTests.swift

```diff
@@ -462,6 +471,58 @@ class TestCodable : TestCodableSuper {
         }
     }
 
+    func test_Dictionary_JSON() {
+        enum X: String, Codable { case a, b }
+        enum Y: String, Codable, CodingKeyRepresentable { case a, b }
+        enum Z: String, Codable, CodingKeyRepresentable {
+            case a
+            case b
+            init?<T: CodingKey>(codingKey: T) {
+                self.init(rawValue: codingKey.stringValue)
+            }
+            var codingKey: CodingKey {
+                GenericCodingKey(stringValue: self.rawValue)
+            }
+        }
+
+        enum U: Int, Codable { case a = 0, b}
+        enum V: Int, Codable, CodingKeyRepresentable { case a = 0, b }
+        enum W: Int, Codable, CodingKeyRepresentable {
+            case a = 0
+            case b
+            init?<T: CodingKey>(codingKey: T) {
+                self.init(rawValue: codingKey.intValue!)
+            }
+            var codingKey: CodingKey {
+                GenericCodingKey(intValue: self.rawValue)
+            }
+        }
+
+
+        struct GenericCodingKey: CodingKey {
+            var stringValue: String
+            var intValue: Int?
+
+            init(stringValue: String) {
+                self.stringValue = stringValue
+            }
+
+            init(intValue: Int) {
+                self.stringValue = "\(intValue)"
+                self.intValue = intValue
+            }
+        }
+
+        expectRoundTripEqualityThroughJSON(for: [X.a: true],             #"["a",true]"#,           lineNumber: #line)
+        expectRoundTripEqualityThroughJSON(for: [Y.a: true, Y.b: false], #"{"a":true,"b":false}"#, lineNumber: #line)
+        expectRoundTripEqualityThroughJSON(for: [Z.a: true, Z.b: false], #"{"a":true,"b":false}"#, lineNumber: #line)
+
+        expectRoundTripEqualityThroughJSON(for: [U.a: true],             #"[0,true]"#,             lineNumber: #line)
+        expectRoundTripEqualityThroughJSON(for: [V.a: true, V.b: false], #"{"0":true,"1":false}"#, lineNumber: #line)
+        expectRoundTripEqualityThroughJSON(for: [W.a: true, W.b: false], #"{"0":true,"1":false}"#, lineNumber: #line)
```

Comment:
Should `Z` and `W` use different keys than the default implementations?

Should the `expectedJSON` parameter have an argument label?

Reply:
Yes and yes. I'll fix that. :-)